src/Runner.Worker/StepsRunner.cs

```diff
@@ -26,6 +21,8 @@ public interface IStep
         string DisplayName { get; set; }
         IExecutionContext ExecutionContext { get; set; }
         TemplateToken Timeout { get; }
+        bool TryUpdateDisplayName(out bool updated);
+        bool EvaluateDisplayName(DictionaryContextData contextData, IExecutionContext context, out bool updated);
```

Comment:
I co-located these in the `IStep` interface because I strongly suspect the latter can be eliminated in a second pass refactor.  Having them side-by-side should make that refactor easier to reason about.